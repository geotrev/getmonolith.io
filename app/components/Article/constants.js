import { Tooltips, Accordions, Collapsibles, Modals, Dropdowns } from "undernet"

export const COMPONENTS = [Tooltips, Accordions, Collapsibles, Modals, Dropdowns]
