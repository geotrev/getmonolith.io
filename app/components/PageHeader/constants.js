export const FOCUSABLE_TABINDEX = "-1"
export const UNFOCUSABLE_TABINDEX = null
