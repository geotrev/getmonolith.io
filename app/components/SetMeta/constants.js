export const titleTemplate = placeholder => `Undernet â€“ ${placeholder}`
