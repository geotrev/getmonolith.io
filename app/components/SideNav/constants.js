import { docPages } from "app/routes"

const {
  introduction,
  download,
  css,
  javascript,
  grid,
  typography,
  buttons,
  forms,
  modals,
  collapsibles,
  accordions,
  dropdowns,
  tooltips,
  alignment,
  offsetOrder,
  text,
  display,
  color,
  spacing,
} = docPages

export const NAV_DATA = [
  {
    header: "Overview",
    links: [
      { name: "Introduction", url: introduction },
      { name: "Download", url: download },
      { name: "CSS", url: css },
      { name: "JavaScript", url: javascript },
    ],
  },
  {
    header: "Layout",
    links: [
      { name: "Flex Grid", url: grid },
      { name: "Alignment", url: alignment },
      { name: "Offset / Order", url: offsetOrder },
      { name: "Spacing", url: spacing },
    ],
  },
  {
    header: "Elements",
    links: [
      { name: "Buttons", url: buttons },
      { name: "Forms", url: forms },
      { name: "Typography", url: typography },
    ],
  },
  {
    header: "Components",
    links: [
      { name: "Collapsibles", url: collapsibles },
      { name: "Accordions", url: accordions },
      { name: "Dropdowns", url: dropdowns },
      { name: "Modals", url: modals },
      { name: "Tooltips", url: tooltips },
    ],
  },
  {
    header: "Utilities",
    links: [
      { name: "Color", url: color },
      { name: "Display", url: display },
      { name: "Text", url: text },
    ],
  },
]
