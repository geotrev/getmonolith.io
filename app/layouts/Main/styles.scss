@charset "UTF-8";
@import "app/global";

// .#{$scope} {
@import "src/scss/utilities/reset";
@import "src/scss/utilities/classes";

@import "src/scss/layout/grid";

@import "src/scss/elements/typography";
@import "src/scss/elements/button";
@import "src/scss/elements/form";

@import "src/scss/components/dropdown";
@import "src/scss/components/modal";
@import "src/scss/components/accordion";
@import "src/scss/components/tooltip";
// }

@import "src/scss/utilities/scope-overrides";

// Layout styles:

main {
  display: block;
  @import "prism";
}

.has-feather {
  display: inline-flex;
  align-items: center;

  svg {
    margin-left: 5px;
  }
}
