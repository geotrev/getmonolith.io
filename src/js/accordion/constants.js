export const Selectors = {
  // unique
  DATA_ACCORDION: "data-accordion",
  DATA_COLLAPSIBLE: "data-collapsible",
  DATA_TOGGLE_MULTIPLE: "data-toggle-multiple",
  // common
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_HIDDEN: "aria-hidden",
}

export const Messages = {
  NO_ACCORDION_ID_ERROR:
    "Could not initialize accordion; you must include a value for the 'data-accordion' attribute.",
  NO_ACCORDION_ERROR: id => `Could not find element matching [data-accordion='${id}']`,
  TRIGGERS_TO_COLLAPSIBLES_LENGTH_WARNING: (id, triggersLength, collapsiblesLength) =>
    `Your accordion with id '${id}' has ${triggersLength} triggers and ${collapsiblesLength} collapsibles; make sure there is a trigger for each collapsible!`,
}
