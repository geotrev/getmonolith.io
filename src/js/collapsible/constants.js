export const Selectors = {
  // unique
  DATA_COLLAPSIBLE: "data-collapsible",
  // common
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_CONTROLS: "aria-controls",
  ARIA_HIDDEN: "aria-hidden",
  ARIA_LABELLEDBY: "aria-labelledby",
  TABINDEX: "tabindex",
  // classes
  TRIGGER_READY_CLASS: "collapsible-trigger-ready",
  CONTENT_READY_CLASS: "collapsible-content-ready",
}

export const CssProperties = {
  HEIGHT: "height",
  VISIBILITY: "visibility",
}

export const CssValues = {
  AUTO: "auto",
  HIDDEN: "hidden",
  VISIBLE: "visible",
}

export const Events = {
  CLICK: "click",
  TRANSITIONEND: "transitionend",
}

export const Messages = {
  NO_COLLAPSIBLE_ID_ERROR:
    "Could not initialize collapsible; you must include a value for the 'data-collapsible' attribute.",
  NO_TRIGGER_ERROR: id =>
    `Could not find collapsible trigger with [data-target='${id}']; you can't have a collapsible without a trigger.`,
  NO_TRIGGER_ID_ERROR: id => `Could not find id on collapsible trigger with [data-target='${id}'].`,
  NO_CONTENT_ERROR: id =>
    `Could not find collapsible content with id '${id}'; you can't have a collapsible without content.`,
}
