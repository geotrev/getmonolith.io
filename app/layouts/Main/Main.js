import React from "react"
import { Switch, Route } from "react-router-dom"
import { ContextUtil } from "undernet"

import { rootPath, docsPath } from "app/routes"
import GlobalNav from "app/components/GlobalNav"
import Footer from "app/components/Footer"
import PageNotFound from "app/components/PageNotFound"
import Home from "app/pages/Home"
import Docs from "app/pages/Docs"

import "./styles.scss"

import { FOCUSABLE_TABINDEX, UNFOCUSABLE_TABINDEX } from "./constants"

export default class Main extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      headerTabIndex: UNFOCUSABLE_TABINDEX,
      mainTabIndex: UNFOCUSABLE_TABINDEX,
    }

    this.headerRef = React.createRef()
    this.mainRef = React.createRef()
  }

  componentDidMount() {
    ContextUtil.enableFocusOutline()
  }

  handleHeaderFocusClick = event => {
    event.preventDefault()

    this.setState({ headerTabIndex: FOCUSABLE_TABINDEX }, () => {
      this.headerRef.current.focus()
    })
  }

  handleMainFocusClick = event => {
    event.preventDefault()

    this.setState({ mainTabIndex: FOCUSABLE_TABINDEX }, () => {
      this.mainRef.current.focus()
    })
  }

  handleHeaderBlur = () => {
    if (this.state.headerTabIndex === UNFOCUSABLE_TABINDEX) return
    this.setState({ headerTabIndex: UNFOCUSABLE_TABINDEX })
  }

  handleMainBlur = () => {
    if (this.state.mainTabIndex === UNFOCUSABLE_TABINDEX) return
    this.setState({ mainTabIndex: UNFOCUSABLE_TABINDEX })
  }

  render() {
    const { headerTabIndex, mainTabIndex } = this.state

    return (
      <React.Fragment>
        <header
          tabIndex={headerTabIndex}
          ref={this.headerRef}
          onBlur={this.handleHeaderBlur}
          role="banner"
        >
          <GlobalNav handleMainFocusClick={this.handleMainFocusClick} />
        </header>

        <main tabIndex={mainTabIndex} ref={this.mainRef} onBlur={this.handleMainBlur} role="main">
          <Switch>
            <Route exact path={rootPath} component={Home} />
            <Route path={docsPath} component={Docs} />
            <Route component={PageNotFound} />
          </Switch>
        </main>

        <footer>
          <Footer handleHeaderFocusClick={this.handleHeaderFocusClick} />
        </footer>
      </React.Fragment>
    )
  }
}
