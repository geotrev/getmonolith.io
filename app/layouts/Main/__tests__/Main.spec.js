import React from "react"
import { BrowserRouter as Router } from "react-router-dom"
import Main from "../Main"

jest.mock("react-feather/dist/icons/twitter", () => global.simpleMock("Twitter"))
jest.mock("react-feather/dist/icons/github", () => global.simpleMock("Github"))
jest.mock("app/pages/Home", () => global.simpleMock("Home"))
jest.mock("app/pages/Docs", () => global.simpleMock("Docs"))
jest.mock("app/components/PageNotFound", () => global.simpleMock("PageNotFound"))

import { UNFOCUSABLE_TABINDEX } from "../constants"

const mountComponent = () => {
  return mount(
    <Router>
      <Main />
    </Router>
  )
}

describe("<Main />", () => {
  describe("#render", () => {
    it("renders", () => {
      const wrapper = mountComponent()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("Jump to top", () => {
    it("focuses header tag when jump-to-top button is clicked", () => {
      // Given
      const wrapper = mountComponent()
      const header = wrapper.find("header")
      // When
      wrapper.find("footer .is-visually-hidden-focusable").simulate("click")
      // Then
      expect(header.is(":focus")).toBe(true)
    })

    it("removes [tabindex] of header tag when blurred", () => {
      // Given
      const wrapper = mountComponent()
      const header = wrapper.find("header")
      // When
      wrapper.find("footer .is-visually-hidden-focusable").simulate("click")
      header.simulate("blur")
      // Then
      expect(header.props().tabIndex).toEqual(UNFOCUSABLE_TABINDEX)
    })
  })

  describe("Skip to main content", () => {
    it("focuses main content when skip to main content button is clicked", () => {
      // Given
      const wrapper = mountComponent()
      const main = wrapper.find("main")
      // When
      wrapper.find("header .is-visually-hidden-focusable").simulate("click")
      // Then
      expect(main.is(":focus")).toBe(true)
    })

    it("removes [tabindex] of main tag when blurred", () => {
      // Given
      const wrapper = mountComponent()
      const main = wrapper.find("main")
      // When
      wrapper.find("footer .is-visually-hidden-focusable").simulate("click")
      main.simulate("blur")
      // Then
      expect(main.props().tabIndex).toEqual(UNFOCUSABLE_TABINDEX)
    })
  })
})
