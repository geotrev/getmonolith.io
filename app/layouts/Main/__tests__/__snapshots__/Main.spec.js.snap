// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> #render renders 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Main>
      <header
        onBlur={[Function]}
        role="banner"
      >
        <GlobalNav
          handleRefocusClick={[Function]}
          mainRef={
            Object {
              "current": <main
                role="main"
              />,
            }
          }
        >
          <nav
            className="is-fluid grid"
            id="global-nav"
          >
            <ul
              className="nav-list row has-no-padding"
            >
              <li
                className="is-small-5 is-xsmall-12 column"
                role="none"
              >
                <Link
                  className="logo"
                  role="menuitem"
                  to="/"
                >
                  <a
                    className="logo"
                    href="/"
                    onClick={[Function]}
                    role="menuitem"
                  >
                    <img
                      alt="Undernet"
                      src={Object {}}
                    />
                  </a>
                </Link>
              </li>
              <li
                className="is-small-7 is-xsmall-12 column"
              >
                <ul
                  className="row"
                >
                  <li
                    role="none"
                  >
                    <button
                      className="is-visually-hidden-focusable"
                      onClick={[Function]}
                      role="menuitem"
                      type="button"
                    >
                      Skip to main content
                    </button>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      className="has-feather"
                      href="https://www.twitter.com/gwtrev"
                      role="menuitem"
                    >
                      <Twitter
                        focusable="false"
                        role="presentation"
                      />
                      <span
                        className="is-visually-hidden"
                      >
                        Open link to www.twitter.com/gwtrev
                      </span>
                    </a>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      className="has-feather"
                      href="https://www.github.com/geotrev/undernet"
                      role="menuitem"
                    >
                      <Github
                        focusable="false"
                        role="presentation"
                      />
                      <span
                        className="is-visually-hidden"
                      >
                        Open link to www.github.com/geotrev/undernet
                      </span>
                    </a>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      href="/docs/overview/introduction"
                      role="menuitem"
                    >
                      Documentation
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </GlobalNav>
      </header>
      <main
        onBlur={[Function]}
        role="main"
      >
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <Home
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            />
          </Route>
        </Switch>
      </main>
      <footer>
        <Footer
          handleRefocusClick={[Function]}
          headerRef={
            Object {
              "current": <header
                role="banner"
              >
                <nav
                  class="is-fluid grid"
                  id="global-nav"
                >
                  <ul
                    class="nav-list row has-no-padding"
                  >
                    <li
                      class="is-small-5 is-xsmall-12 column"
                      role="none"
                    >
                      <a
                        class="logo"
                        href="/"
                        role="menuitem"
                      >
                        <img
                          alt="Undernet"
                          src="[object Object]"
                        />
                      </a>
                    </li>
                    <li
                      class="is-small-7 is-xsmall-12 column"
                    >
                      <ul
                        class="row"
                      >
                        <li
                          role="none"
                        >
                          <button
                            class="is-visually-hidden-focusable"
                            role="menuitem"
                            type="button"
                          >
                            Skip to main content
                          </button>
                        </li>
                        <li
                          role="none"
                        >
                          <a
                            class="has-feather"
                            href="https://www.twitter.com/gwtrev"
                            role="menuitem"
                          >
                            <span
                              class="is-visually-hidden"
                            >
                              Open link to www.twitter.com/gwtrev
                            </span>
                          </a>
                        </li>
                        <li
                          role="none"
                        >
                          <a
                            class="has-feather"
                            href="https://www.github.com/geotrev/undernet"
                            role="menuitem"
                          >
                            <span
                              class="is-visually-hidden"
                            >
                              Open link to www.github.com/geotrev/undernet
                            </span>
                          </a>
                        </li>
                        <li
                          role="none"
                        >
                          <a
                            href="/docs/overview/introduction"
                            role="menuitem"
                          >
                            Documentation
                          </a>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </nav>
              </header>,
            }
          }
        >
          <div
            className="footer-wrapper is-small-section is-fluid grid"
          >
            <div
              className="row"
            >
              <div
                className="has-no-padding column"
              >
                <p
                  className="has-no-margin-block-end has-text-center has-gray900-text"
                >
                  Undernet is a front-end framework created and maintained by
                   
                  <a
                    className="has-white-text"
                    href="http://www.geotrev.com"
                  >
                    George Treviranus
                  </a>
                  .
                </p>
                <p
                  className="has-no-margin-block-end has-text-center has-gray900-text"
                >
                  Animations provided by
                   
                  <a
                    className="has-white-text"
                    href="https://folgert.com/"
                  >
                    Gavin Folgert
                  </a>
                  . ❤️
                </p>
                <ul
                  className="has-text-center has-no-padding has-gray900-text has-unstyled-list has-direction-row has-display-flex has-justify-content-center"
                >
                  <li
                    className="has-no-padding"
                    role="none"
                  >
                    <a
                      className="has-white-text has-feather"
                      href="https://www.twitter.com/gwtrev"
                      role="menuitem"
                    >
                      <Twitter
                        focusable="false"
                        role="presentation"
                      />
                      <span
                        className="is-visually-hidden"
                      >
                        Open link to www.twitter.com/gwtrev
                      </span>
                    </a>
                  </li>
                  <li
                    className="has-no-padding"
                    role="none"
                  >
                    <a
                      className="has-white-text has-feather"
                      href="https://www.github.com/geotrev/undernet"
                      role="menuitem"
                    >
                      <Github
                        focusable="false"
                        role="presentation"
                      />
                      <span
                        className="is-visually-hidden"
                      >
                        Open link to www.github.com/geotrev/undernet
                      </span>
                    </a>
                  </li>
                  <li
                    className="has-no-padding"
                    role="none"
                  >
                    <button
                      className="is-visually-hidden-focusable"
                      onClick={[Function]}
                      role="menuitem"
                      type="button"
                    >
                      Return to top of page
                    </button>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </Footer>
      </footer>
    </Main>
  </Router>
</BrowserRouter>
`;
